plugins/source/aws/client/client.go

```diff
@@ -287,12 +285,26 @@ func configureAwsClient(ctx context.Context, logger zerolog.Logger, awsConfig *S
 				opts.RoleSessionName = account.RoleSessionName
 			})
 		}
+
 		if stsClient == nil {
 			stsClient = sts.NewFromConfig(awsCfg)
 		}
 		provider := stscreds.NewAssumeRoleProvider(stsClient, account.RoleARN, opts...)
 
-		awsCfg.Credentials = aws.NewCredentialsCache(provider)
+		awsCfg.Credentials = aws.NewCredentialsCache(provider, func(options *aws.CredentialsCacheOptions) {
+			// ExpiryWindow will allow the credentials to trigger refreshing prior to
+			// the credentials actually expiring. This is beneficial so race conditions
+			// with expiring credentials do not cause requests to fail unexpectedly
+			// due to ExpiredToken exceptions.
+			//
+			// An ExpiryWindow of 1 minute would cause calls to IsExpired() to return true
+			// 1 minute before the credentials are actually expired. This can cause an
+			// increased number of requests to refresh the credentials to occur.
+			options.ExpiryWindow = 1 * time.Minute
```

Comment:
```suggestion
			// ExpiryWindow will allow the credentials to trigger refreshing prior to
			// the credentials actually expiring. This is beneficial so race conditions
			// with expiring credentials do not cause requests to fail unexpectedly
			// due to ExpiredToken exceptions.
			//
			// An ExpiryWindow of 5 minute would cause calls to IsExpired() to return true
			// 5 minutes before the credentials are actually expired. This can cause an
			// increased number of requests to refresh the credentials to occur. We balance this with jitter.
			options.ExpiryWindow = 5 * time.Minute
```